src/main/java/io/confluent/kafkarest/converters/JsonNodeConverters.java

```diff
@@ -0,0 +1,79 @@
+package io.confluent.kafkarest.converters;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.*;
+import org.apache.avro.Schema;
+
+import java.math.BigDecimal;
+import java.util.Iterator;
+import java.util.Objects;
+
+public class JsonNodeConverters {
+
+    static JsonNode textToDecimal(JsonNode jsonNode, Schema schema) {
+        if (jsonNode instanceof TextNode){
+            String encodedBigDecimal = jsonNode.asText();
+            int scale = schema.getJsonProps().get("scale").asInt();
+            BigDecimal bd = BigDecimalDecoder.fromEncodedString(encodedBigDecimal, scale);
+            return new DecimalNode(bd);
+        } else {
+            return jsonNode;
+        }
+    };
+
+    static JsonNode decimalToText(JsonNode jsonNode, Schema schema) {
+        if (jsonNode instanceof DoubleNode) {
```

Comment:
as a suggestion, I would make these methods more conditional in their names. For example, if the method was `isDecimal` (which encapsulated your `instanceOf` logic), then as a result of that condition being true, you'd do your conversion. False would lead to the function being an identity on the input `jsonNode`. The benefit of this is in `AvroConverter`, the code isn't read as "I am definitely turning this into text from a decimal", but more so "I will convert this into text iff it's a decimal". The same can be said for `textToDecimal`.